xline/src/server/kv_server.rs

```diff
@@ -310,19 +309,34 @@ where
             self.kv_storage.compacted_revision(),
             self.kv_storage.revision(),
         )?;
-        let is_fast_path = false; // lock need revision of txn
-        let (cmd_res, sync_res) = self.propose(request, is_fast_path).await?;
 
-        let mut res = Self::parse_response_op(cmd_res.decode().into());
-        if let Some(sync_res) = sync_res {
-            let revision = sync_res.revision();
-            debug!("Get revision {:?} for TxnRequest", revision);
-            Self::update_header_revision(&mut res, revision);
-        }
+        let res = if txn_req.is_read_only() {
+            debug!("TxnRequest is read only");
+            let is_serializable = txn_req.is_serializable();
+            let token = get_token(request.metadata());
+            let wrapper = RequestWithToken::new_with_token(request.into_inner().into(), token);
+            let propose_id = self.generate_propose_id();
+            let cmd = command_from_request_wrapper::<S>(propose_id, wrapper, None);
+            if !is_serializable {
+                self.wait_read_state(&cmd).await?;
+            }
+            self.do_serializable(cmd.request())?
+        } else {
+            let is_fast_path = false; // lock need revision of txn
```

Comment:
after #391 the revision will be returned in execution result, so we can use fast path here for better performance

Reply:
Fixed